### PR TITLE
fix NPE on LoadFilesListTask#listTrashBinFiles

### LINKED ISSUE #4142: NPE on `LoadFilesListTask#listTrashBinFiles`
## Issue explanation (write below this line)
Amaze breaks when accessing trash bin.

## Exception
* __App Name:__ Amaze File Manager
* __Package:__ com.amaze.filemanager
* __Version:__ 3.10
* __User Action:__ UI Error
* __Request:__ Application crash
* __OS:__ Linux Android 14 - 34
* __Device:__ panther
* __Model:__ Pixel 7
* __Product:__ panther
<details><summary><b>Crash log </b></summary><p>

```
java.lang.RuntimeException: An error occurred while executing doInBackground()
at android.os.AsyncTask$4.done(AsyncTask.java:415)
at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:381)
at java.util.concurrent.FutureTask.setException(FutureTask.java:250)
at java.util.concurrent.FutureTask.run(FutureTask.java:269)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
at java.lang.Thread.run(Thread.java:1012)
Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'java.util.Iterator java.util.List.iterator()' on a null object reference
at com.amaze.filemanager.asynchronous.asynctasks.LoadFilesListTask.listTrashBinFiles(LoadFilesListTask.java:607)
at com.amaze.filemanager.asynchronous.asynctasks.LoadFilesListTask.getCachedMediaList(LoadFilesListTask.java:286)
at com.amaze.filemanager.asynchronous.asynctasks.LoadFilesListTask.doInBackground(LoadFilesListTask.java:169)
at com.amaze.filemanager.asynchronous.asynctasks.LoadFilesListTask.doInBackground(LoadFilesListTask.java:91)
at android.os.AsyncTask$3.call(AsyncTask.java:394)
at java.util.concurrent.FutureTask.run(FutureTask.java:264)
... 3 more

```
</details>
<hr>

### PR DESCRIPTION
**Pull request by malirauf**
## Description
NPE on LoadFilesListTask#listTrashBinFiles

#### Issue tracker   
Fixes #4142

### CODE REVIEW

**Review by ZweDen051228 — APPROVED** [review]

**Review by ZweDen051228 — APPROVED** [review]

**Comment by ZEYER45** [comment]
Well become...the house...




FRÉDÉRICK ZEYER

Le mer. 24 juil. 2024, 16:33, ZweDen051228 ***@***.***> a
écrit :

File: app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/LoadFilesListTask.java [hunk 1]
```diff
@@ -605,24 +605,29 @@ else if (cursor.getCount() > 0 && cursor.moveToFirst()) {
...
-            new HybridFile(
-                OpenMode.TRASH_BIN,
-                trashBinFile.getDeletedPath(
-                    AppConfig.getInstance().getTrashBinInstance().getConfig()),
-                trashBinFile.getFileName(),
-                trashBinFile.isDirectory());
-        if (trashBinFile.getDeleteTime() != null) {
-          hybridFile.setLastModified(trashBinFile.getDeleteTime() * 1000);
+
+      List<TrashBinFile> filesInBin = trashBin.listFilesInBin();
```
Comment (VishnuSanal) [inline app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/LoadFilesListTask.java]: Please do `if ... return;`. That looks much cleaner & less changes.

**Review by VishnuSanal — CHANGES_REQUESTED** [review]

**Review by VishnuSanal — COMMENTED** [review]
@malirauf thanks for your interest in contributing to Amaze! I've some changes to be done, please see the reviews.

File: app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/LoadFilesListTask.java (hunk 1 above)
Comment (malirauf) [reply app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/LoadFilesListTask.java]: Yes, you are right.

**Review by VishalNehra — APPROVED** [review]

File: app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java [hunk 2]
```diff
@@ -1863,7 +1863,7 @@ public TrashBinFile toTrashBinRestoreFile(Context context) {
     TrashBin trashBin = AppConfig.getInstance().getTrashBinInstance();
     for (TrashBinFile trashBinFile : trashBin.listFilesInBin()) {
       if (trashBinFile.getDeletedPath(trashBin.getConfig()).equals(path)) {
-        // finding path to restore to
+        // finding path to restore tof
```
Comment (VishnuSanal) [inline app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java:1866]: please revert

**Review by VishnuSanal — APPROVED** [review]
LGTM 🚀

**Pull request merged**